components/script/dom/blob.rs

```diff
@@ -110,13 +108,7 @@ impl Blob {
             }
             BlobImpl::Sliced(ref grandparent, ref old_rel_pos) => {
                 // Adjust the slicing position, using same parent
-                let new_rel_pos = old_rel_pos.slice_inner(&rel_pos);
-
-                if let BlobImpl::File(ref f) = *grandparent.blob_impl.borrow() {
-                    inc_ref_id(global.r(), f.id.clone());
-                }
-
-                BlobImpl::Sliced(grandparent.clone(), new_rel_pos)
+                BlobImpl::Sliced(grandparent.clone(), old_rel_pos.slice_inner(&rel_pos))
```

Comment:
In all cases above, we created a `Sliced`, i.e. not directly holding the `FileID`, thus not contributing to the ref-count.